### ISSUE TITLE
[explicit-member-accessibility] add @internal JSDoc comment as valid accessibility modifier

### ISSUE THREAD

**Issue by htho** [typescript-eslint/typescript-eslint#1215]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": ["error"]
  }
}
```

```TS
class Test {
  private a;
  protected b;
  /** @internal */
  c;
  public d;
}
```

**Expected Result**
`c` should not have an error, because `internal` should be treated as a valid modifier.

**Actual Result**
`c` has an error, because this rule does not care about the `internal` JSDoc comment.

**Additional Info**
I want the minifier (terser) to minify all non-public members. This is possible in combination with the `member-naming` rule (#1215). It will be configured to enforce non-public members to be prefixed with an underscore. Terser can be told to minify properties prefixed with an underscore.

**Comment by bradzacher** [comment]
Why not just annotate it as public?

**Comment by htho** [comment]
> Why not just annotate it as public?

I think it is a taste thing. For me, there should only be one modifier.

I learned OOP with Java, where no modifier means [package public](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html). A fixer that automatically marks any properties without a modifier as internal would be great for my use-case and understanding.

**Comment by bradzacher** [comment]
the problem is that there is no existing jsdoc parsing code within this plugin.
So to make this work we'd have to create that, and make sure comment attachment works correctly.

Happy to accept a PR for it. But it's a rather large piece of work, which you'd have to do yourself.

**Comment by jimmywarting** [comment]
IMO i think public/private should stop being used, instead private class field/method should be used (`#`) and everything that dose not have it should be public. as simple as that...

**Comment by JoshuaKGoldberg** [comment]
It's been several years and nobody has tackled this very large piece of work. And `#` privates are indeed the canonical right way to do privacy. 

If someone _really_ wants to take this on, please let us know - but I'm going to close this issue as wontfix for now. Thanks all!
